#! /bin/sh /usr/share/dpatch/dpatch-run
## r213 | relnev | 2003-02-14 06:51:57 +0100 (Fr, 14 Feb 2003) | 2 lines
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: added some missing include guards

@DPATCH@
diff -urNad rott-1.0+dfsg~/_rt_util.h rott-1.0+dfsg/_rt_util.h
--- rott-1.0+dfsg~/_rt_util.h	2008-01-09 13:52:28.000000000 +0100
+++ rott-1.0+dfsg/_rt_util.h	2008-01-10 09:21:04.000000000 +0100
@@ -20,10 +20,12 @@
 #ifndef _rt_util_private
 #define _rt_util_private
 
+#ifdef DOS
 #define PEL_WRITE_ADR   0x3c8
 #define PEL_READ_ADR    0x3c7
 #define PEL_DATA        0x3c9
 #define PEL_MASK        0x3c6
+#endif
 
 #define ERRORROW        2
 #define ERRORCOL        11
diff -urNad rott-1.0+dfsg~/byteordr.h rott-1.0+dfsg/byteordr.h
--- rott-1.0+dfsg~/byteordr.h	2008-01-10 09:21:02.000000000 +0100
+++ rott-1.0+dfsg/byteordr.h	2008-01-10 09:21:04.000000000 +0100
@@ -23,6 +23,9 @@
 //
 //***************************************************************************
 
+#ifndef BYTEORDER_H
+#define BYTEORDER_H
+
 typedef void (*converter_t)(void *, int);
 
 #define DECLARE_CONVERTER(type) void Cvt_##type(void *lmp, int num);
@@ -36,3 +39,5 @@
 DECLARE_CONVERTER(cfont_t);
 void CvtNull(void *lmp, int num);
 converter_t CvtForType(int type);
+
+#endif
diff -urNad rott-1.0+dfsg~/cin_glob.h rott-1.0+dfsg/cin_glob.h
--- rott-1.0+dfsg~/cin_glob.h	2008-01-09 13:52:28.000000000 +0100
+++ rott-1.0+dfsg/cin_glob.h	2008-01-10 09:21:04.000000000 +0100
@@ -22,6 +22,9 @@
 #include "isr.h"
 #include <time.h>
 
+#ifndef CIN_GLOB_H
+#define CIN_GLOB_H
+
 #define DUMP 0
 
 #define CLOCKSPEED (VBLCOUNTER)
@@ -31,3 +34,4 @@
 boolean CinematicAbort( void );
 void ClearCinematicAbort( void );
 
+#endif
diff -urNad rott-1.0+dfsg~/modexlib.c rott-1.0+dfsg/modexlib.c
--- rott-1.0+dfsg~/modexlib.c	2008-01-10 09:21:02.000000000 +0100
+++ rott-1.0+dfsg/modexlib.c	2008-01-10 09:21:04.000000000 +0100
@@ -389,6 +389,8 @@
 
 #include "SDL.h"
 
+#ifndef STUB_FUNCTION
+
 /* rt_def.h isn't included, so I just put this here... */
 #if !defined(ANSIESC)
 #define STUB_FUNCTION fprintf(stderr,"STUB: %s at " __FILE__ ", line %d, thread %d\n",__FUNCTION__,__LINE__,getpid())
@@ -396,6 +398,8 @@
 #define STUB_FUNCTION
 #endif
 
+#endif
+
 /*
 ====================
 =
diff -urNad rott-1.0+dfsg~/modexlib.h rott-1.0+dfsg/modexlib.h
--- rott-1.0+dfsg~/modexlib.h	2008-01-10 09:21:02.000000000 +0100
+++ rott-1.0+dfsg/modexlib.h	2008-01-10 09:21:04.000000000 +0100
@@ -40,6 +40,7 @@
 //
 //***************************************************************************
 
+#ifdef DOS
 #define SC_INDEX                0x3C4
 #define SC_DATA                 0x3C5
 #define SC_RESET                0
@@ -100,7 +101,7 @@
 #define PEL_WRITE_ADR           0x3c8
 #define PEL_READ_ADR            0x3c7
 #define PEL_DATA                0x3c9
-
+#endif
 
 extern  int      ylookup[MAXSCREENHEIGHT];      // Table of row offsets
 extern  int      linewidth;
diff -urNad rott-1.0+dfsg~/rottnet.h rott-1.0+dfsg/rottnet.h
--- rott-1.0+dfsg~/rottnet.h	2008-01-09 13:52:28.000000000 +0100
+++ rott-1.0+dfsg/rottnet.h	2008-01-10 09:21:04.000000000 +0100
@@ -23,10 +23,12 @@
 
 #include "develop.h"
 
+#ifdef DOS
 #define PEL_WRITE_ADR   0x3c8
 #define PEL_DATA        0x3c9
 
 #define I_ColorBlack(r,g,b) {outp(PEL_WRITE_ADR,0);outp(PEL_DATA,r);outp(PEL_DATA,g);outp(PEL_DATA,b);};
+#endif
 
 #define	MAXNETNODES		14			// max computers in a game
 
diff -urNad rott-1.0+dfsg~/rt_def.h rott-1.0+dfsg/rt_def.h
--- rott-1.0+dfsg~/rt_def.h	2008-01-10 09:21:02.000000000 +0100
+++ rott-1.0+dfsg/rt_def.h	2008-01-10 09:21:04.000000000 +0100
@@ -248,9 +248,6 @@
 #define MaxKbds            2
 #define MaxJoys            2
 #define NumCodes           128
-#define MReset             0
-#define MButtons           3
-#define MDelta             11
 
 // Key definitions
 
diff -urNad rott-1.0+dfsg~/rt_util.c rott-1.0+dfsg/rt_util.c
--- rott-1.0+dfsg~/rt_util.c	2008-01-10 09:21:03.000000000 +0100
+++ rott-1.0+dfsg/rt_util.c	2008-01-10 09:21:04.000000000 +0100
@@ -928,7 +928,7 @@
     {
         *ptr = '\0';
         f->dir = opendir(f->pattern);
-        memmove(f->pattern, ptr + 1, strlen(ptr) + 1);
+        memmove(f->pattern, ptr + 1, strlen(ptr + 1) + 1);
     }
 
     return(_dos_findnext(f));
diff -urNad rott-1.0+dfsg~/version.h rott-1.0+dfsg/version.h
--- rott-1.0+dfsg~/version.h	2008-01-09 13:52:28.000000000 +0100
+++ rott-1.0+dfsg/version.h	2008-01-10 09:21:04.000000000 +0100
@@ -17,6 +17,12 @@
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
 */
+
+#ifndef VERSION_H
+#define VERSION_H
+
 #define ROTTMAJORVERSION 1
 #define ROTTMINORVERSION 4
 #define ROTTVERSION ((ROTTMAJORVERSION*10)+(ROTTMINORVERSION))
+
+#endif
